server/service/integration_android_software_test.go

```diff
@@ -221,4 +222,81 @@ func (s *integrationMDMTestSuite) TestAndroidAppSelfService() {
 	// Should have hit the android API endpoint
 	s.Assert().True(s.androidAPIClient.EnterprisesPoliciesModifyPolicyApplicationsFuncInvoked)
 
+	// Test Android app configurations
+
+	// Android app with configuration
+	appConfiguration := json.RawMessage(`{"workProfileWidgets": "WORK_PROFILE_WIDGETS_ALLOWED"}`)
+	androidAppWithConfig := &fleet.VPPApp{
+		VPPAppTeam: fleet.VPPAppTeam{
+			VPPAppID: fleet.VPPAppID{
+				AdamID:   "com.fooooooo",
+				Platform: fleet.AndroidPlatform,
+			},
+			Configuration: appConfiguration,
+		},
+		Name:             "foo",
+		BundleIdentifier: "com.fooooooo",
+		IconURL:          "https://example.com/images/2",
+	}
+
+	// Add Android app
+	var appWithConfigResp addAppStoreAppResponse
+	s.DoJSON(
+		"POST",
+		"/api/latest/fleet/software/app_store_apps",
+		&addAppStoreAppRequest{
+			AppStoreID:    androidAppWithConfig.AdamID,
+			Platform:      androidAppWithConfig.VPPAppID.Platform,
+			Configuration: androidAppWithConfig.Configuration,
+		},
+		http.StatusOK,
+		&appWithConfigResp,
+	)
+
+	// Verify that activity includes configuration
+	s.lastActivityMatches(fleet.ActivityAddedAppStoreApp{}.ActivityName(),
```

Comment:
There are some changes to display names logic that might cause these tests to fail. Probably worth merging `main` once #36219 lands.